pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -1531,6 +1531,70 @@ class Timestamp(_Timestamp):
             ) from err
         return _dt.ctime()
 
+    def date(self):
+        """
+        Return date object with same year, month and day.
+
+        Examples
+        --------
+        >>> ts = pd.Timestamp('2023-01-01 10:00:00.00')
+        >>> ts
+        Timestamp('2023-01-01 10:00:00')
+        >>> ts.date()
+        datetime.date(2023, 1, 1)
+        """
+        try:
+            _dt = datetime(self.year, self.month, self.day,
+                           self.hour, self.minute, self.second,
+                           self.microsecond, self.tzinfo, fold=self.fold)
+        except ValueError as err:
+            raise NotImplementedError(
+                "date not yet supported on Timestamps which "
+                "are outside the range of Python's standard library. "
+                "For now, please call the components you need (such as `.year` "
+                "and `.month`) and construct your string from there."
+            ) from err
+        return _dt.date()
+
+    def dst(self):
+        """
+        Return self.tzinfo.dst(self).
+
+        Examples
+        --------
+        >>> ts = pd.Timestamp('2000-06-01 00:00:00',
+        ...                   tz='Europe/Brussels').dst()
+        >>> ts
+        datetime.timedelta(seconds=3600)
+        """
+        _dt = datetime(self.year, self.month, self.day,
+                       self.hour, self.minute, self.second,
+                       self.microsecond, self.tzinfo, fold=self.fold)
+        return _dt.dst()
```

Comment:
looks like we're not catching anything here, so can we just return `return super().dst()`?

maybe we can do the same for the others where we're just overriding the docstrings, like
```python
       try:
            return super().isocalendar()
        except ValueError as err:
            raise NotImplementedError(
                "isocalendar not yet supported on Timestamps which "
                "are outside the range of Python's standard library. "
            ) from err
```
and similarly for the others

(in a separate PR, we could also try to do the same trick for the other items where we're overriding the docstrings, like strftime - admittedly I should've thought of that when I made that PR)